### DISCUSSION TITLE
【模型压缩推全计划】为六大套件新增模型压缩功能

### DISCUSSION THREAD

**Discussion by shiyutang** [PaddlePaddle/PaddleOCR#12736] [Ideas & Features]
## 背景
目前各套件的模型压缩能力参差不齐，而模型压缩作为部署之前的一步，可以在不损害或者少量损害模型精度的情况下，对模型的能耗，速度、大小都有显著的改善。因此为了对各套件的模型压缩进行推全，我们提出了基于PaddleSlim的ACT为各大套件新增模型压缩功能的计划。

## 解决步骤：
1. 查看[Paddleslim的ACT文档](https://github.com/PaddlePaddle/PaddleSlim/tree/develop/example/auto_compression)，学习ACT，熟悉基于不同任务进行ACT中量化压缩的主要步骤。
2. 为某一任务接入ACT中的量化训练压缩功能：在适配paddle2.5，paddleslim2.5和套件develop版本的前提下，仿照[pp-Liteseg接入ACT的PR](https://github.com/PaddlePaddle/PaddleSeg/pull/3457)，在各大套件的部署文件夹下（例如分割为PaddleSeg/deploy/slim/act）  接入ACT。下面以分割为例说明需要包括的新增文件内容：
     - 增加ACT启动脚本 run_seg.py 在 PaddleSeg/deploy/slim/act/run_seg.py
     - 增加量化训练配置文件在 PaddleSeg/deploy/slim/act/configs/ppliteseg_qat.py
     - 增加用于验证精度和速度的推理脚本在PaddleSeg/deploy/slim/act/test_seg.py

5. 参考paddleslim的ACT使用文档，对量化训练模型进行导出，并验证不同模型在量化蒸馏训练前后的精度和速度对比，验证通过后，撰写量化训练使用文档，增加相关Readme在 PaddleSeg/deploy/slim/act/readme.md。
6. 提交PR到对应模型的仓库中。

注：不同套件的部署目录如下，将上述分割的目录进行相应替换即可：
  - PaddleClas： PaddleClas/deploy/slim/act
  - PaddleDetection： PaddleDetection/deploy/auto_compression(该目录下的文件为之前接入的文件，可以复用)
  - Paddle3D：Paddle3D/deploy/slim_act
  - PaddleOCR: PaddleOCR/deploy/slim/act
  - PaddleSeg: PaddleSeg/deploy/slim/act/

### 同一个任务仅有配置文件差异，一个任务需要完成对应的模型配置，领取通用目标检测任务的开发者将认领四个检测模型。

任务名称 | 模型参数名称 | 报名开发者
-- | -- | --
通用目标检测 | DINO_r50_4scale | @MINGtoMING
通用目标检测 | RT-DETR-L | @MINGtoMING
通用目标检测 | PP-YOLOE+_crn_l_80e | @MINGtoMING
通用目标检测 | PP-PicoDet_s_320_LCNet | @MINGtoMING
通用图像分类 | SwinTransformer_base_patch4_window7_224 | @vax521
通用图像分类 | PP-LCNet_x1_0 | @vax521
通用图像分类 | CLIP_vit_base_patch16_224 | @vax521
通用图像分类 | PP-HGNet_small | @vax521
通用图像分类 | ResNet50 | @vax521
通用图像分类 | MobileNetV3_small_x1_0 | @vax521
通用文本检测 | PP-OCRv4-server-det | @xu-peng-7 
通用文本检测 | PP-OCRv4-mobile-det | @xu-peng-7 
通用文本识别 | PP-OCRv4-server-rec | 
通用文本识别 | PP-OCRv4-mobile-rec |  
通用图像分割 | OCRNet_HRNetW48 | @marshall-dteach 
通用图像分割 | SegFormer-B0 | @marshall-dteach 
通用图像分割 | PP-LiteSeg-Tiny | @marshall-dteach 
通用图像分割 | PP-MobileSeg-Base | @marshall-dteach 
3D目标检测 | CenterPoint_pillars_02voxel | 
3D目标检测 | PETRv2_v99 | 
3D目标检测 | PETRv1_v99 | 
3D目标检测 | CaDDN_ocrnet_hrnet_w18 | 
通用主体检测 | PP-ShiTuv2-det |  @Zheng-Bicheng
通用检索特征 | PP-ShiTuv2-rec |  @Zheng-Bicheng
通用版面分析 | PP-Structurev2-layout | 
通用表格识别 | PP-Structurev2-SLANet | 
通用关键信息抽取 | PP-Structurev2-vi-layoutxlm |

**Comment by WilliamQf-AI** [comment]
队伍名：风清扬
队伍成员微信昵称：WilliamQf-AI
功能描述：通用版面分析

**Comment by WilliamQf-AI** [comment]
队伍名：风清扬
队伍成员微信昵称：WilliamQf-AI
功能描述：通用表格识别

**Comment by ToddBear** [comment]
队伍名：飞桨小队
队伍成员微信昵称：zejia_su
功能描述：通用文本识别

**Comment by marshall-dteach** [comment]
队伍名：德布罗意波
队伍成员微信昵称：TB1250859983
功能描述：通用图像分割

**Comment by xu-peng-7** [comment]
队伍名：Dec20b
队伍成员微信昵称：muted
功能描述：通用文本检测

**Comment by vax521** [comment]
队伍名：晓飞队
队伍成员微信昵称：晓飞
功能描述：通用图像分类

**Comment by Zheng-Bicheng** [comment]
队伍名：Zheng-Bicheng
队伍成员微信昵称：Zheng-Bicheng
功能描述：通用主体检测

**Comment by Zheng-Bicheng** [comment]
队伍名：Zheng-Bicheng
队伍成员微信昵称：Zheng-Bicheng
功能描述：通用检索特征
